packages/menu/src/useMenu.ts

```diff
@@ -635,18 +662,18 @@ export const useMenu = <T extends HTMLElement = HTMLElement, M extends HTMLEleme
    * Respond to clicks outside the  open menu
    */
   useEffect(() => {
-    const win = environment || window;
+    const _document = getDocument(windowProp, documentProp) as Document;
 
     if (controlledIsExpanded) {
-      win.document.addEventListener('keydown', handleMenuKeyDown, true);
+      _document.addEventListener('keydown', handleMenuKeyDown, true);
     } else if (!controlledIsExpanded) {
-      win.document.removeEventListener('keydown', handleMenuKeyDown, true);
+      _document.removeEventListener('keydown', handleMenuKeyDown, true);
     }
 
     return () => {
-      win.document.removeEventListener('keydown', handleMenuKeyDown, true);
+      (_document as Document).removeEventListener('keydown', handleMenuKeyDown, true);
```

Comment:
Is it worth coercing the `getDocument` return value to `Document` so you don't need to repeatedly coerce throughout?